src/peep/peep.c

```diff
@@ -7161,24 +7163,23 @@ static int sub_69A997(sint16 x, sint16 y, uint8 z, uint8 counter, uint16 score,
 	do {
 		if (map_element_get_type(path) != MAP_ELEMENT_TYPE_PATH) continue;
 
-		uint8 slope_dir;
 		if (footpath_element_is_sloped(path) &&
-			(slope_dir = footpath_element_get_slope_direction(path)) != test_edge) {
-			slope_dir ^= 2;
-			if (slope_dir != test_edge) continue;
+			footpath_element_get_slope_direction(path) != test_edge) {
+			if ((footpath_element_get_slope_direction(path) ^ 2) != test_edge) continue;
 			if (path->base_height + 2 != z) continue;
 		} else {
-			if (z != path->base_height) continue;
+			if (path->base_height != z) continue;
 			if (footpath_element_is_wide(path)) continue;
 		}
 
-		if ((!footpath_element_is_queue(path) ||
-			 RCT2_GLOBAL(0x00F1AEE1, uint8) != path->properties.path.ride_index) &&
-			(path->type & RCT2_GLOBAL(0x00F1AEE0, uint8))) continue;
+		if (!footpath_element_is_queue(path) ||
+			RCT2_GLOBAL(0x00F1AEE1, uint8) != path->properties.path.ride_index) {
+			if (path->type & RCT2_GLOBAL(0x00F1AEE0, uint8)) continue;
+		}
 
 		found = true;
 		break;
-	} while (map_element_is_last_for_tile(path++));
+	} while (!map_element_is_last_for_tile(path++));
```

Comment:
This is probably the main cause of issues. Almost every function i write I forget to do this.